core/domain/suggestion_services.py

```diff
@@ -1125,22 +1120,21 @@ def get_reviewable_translation_suggestions_for_single_exp(
     """
     contribution_rights = user_services.get_user_contribution_rights(
         user_id)
+    user_settings = user_services.get_user_settings(user_id)
+    prefered_language_code = user_settings.preferred_translation_language_code
```

Comment:
preferred is misspelled.

Also, what does preferred_translation_language_code (I think that's the language they prefer to use the site in) have to do with anything on the CD?

Reply:
No, the one you are saying we store that using this method https://github.com/oppia/oppia/blob/794b5d78e88f585efadc366b767889596afccacb/core/domain/user_services.py#L1523. Preferred_translation_language_code is the active language chosen in the language filter of the review-dashboard.

Reply:
Hm, are you sure? AFAICT, this is the "language in which the site is shown" in the user's preferences page. I did check that this field is shown to the user in PreferencesHandler in core/controllers/profile.py, and the UI looks like this:
![Screenshot from 2023-08-07 22-48-33](https://github.com/oppia/oppia/assets/10575562/72fbf3c9-919c-449a-8c2e-044a35e2a8ec)

Reply:
The one you are talking about seems to be `preferred_site_language` and the one I am using is `preferred_translation_language_code`.

Reply:
Ah, sorry, my bad. I misread your comment, apologies.

But then, what if the reviewer reviews two languages? Wouldn't the frontend still need to pass in the language we're trying to get the results for?

Reply:
I have added a video recording to demonstrate how this would work for the case you mentioned.


https://github.com/oppia/oppia/assets/98253080/662fb4e6-e059-4590-82ea-a2503e7d573c

Reply:
@Ash-2k3 Thanks, but I think it would also be helpful to get an explanation of *why* it works technically.

Is it the case that, when the new language is chosen, the user settings are updated, and that's why the preferred language code changes? If so, then I suggest that you get that info somewhere else and pass the language code explicitly as an argument into this function. This is because the function name here (which presumably is intended to be generic) says nothing about the user's preferences (and probably shouldn't).

In other words, based on the function's contract (as specified in its name and docstring), this function shouldn't be specific to the user's preferred language. The user is asking for a list of items they are able to review, and we return those items. From a modelling perspective, I think preferences should be dealt with separately so that this lower-level method to be generic for other use cases (or not dealt with at all, and we just pass the language in directly from the frontend or the controller when requesting the relevant results).

Reply:
Okay, I understand it. But then I think we should go with the existing approach here like https://github.com/oppia/oppia/blob/2779e1968460f38f243eb0031c667daaf130f59d/core/storage/suggestion/gae_models.py#L748

Reply:
Yes, that is what I am suggesting. Just pass one language code though, not several.

Reply:
Noticed that the approach which I thought of `preferred_translation_language_code` doesn't work for one case, when the dashboard loads for first time as during that time there's no language selected in language filter.

I think we would want it to be list of language codes, just like we have it in other method. [oppia/core/storage/suggestion/gae_models.py](https://github.com/oppia/oppia/blob/2779e1968460f38f243eb0031c667daaf130f59d/core/storage/suggestion/gae_models.py#L748).

@seanlip, let me know your thoughts please, Thanks.

Reply:
Ahn okay, ig option 1 is better, I would prefer not changing existing workflow for this. Also I have made service method to have language_code as a arg, as it makes sense to make it generic. I am passing the language to this method from the handler, as ig it makes sense that way.